src/App.tsx

```diff
@@ -43,19 +52,29 @@ export function App() {
 
 	return (
 		<Router>
+			<Toaster />
 			<Routes>
-				<Route path="/" element={<Layout />}>
+				<Route path="/" element={<Layout user={user} />}>
 					<Route
-						index
+						path="/"
 						element={
 							<Home data={lists} setListPath={setListPath} user={user} />
 						}
 					/>
-					<Route path="/list" element={<List data={data} />} />
-					<Route
-						path="/manage-list"
-						element={<ManageList listPath={listPath} />}
-					/>
+
+					{/* protected routes */}
+					<Route element={<ProtectRoute user={user} redirectPath="/" />}>
```

Comment:
I'm having a bit of a hard time with this one. On one hand, I see that just about everyone on the www suggests using `Outlet` (or something akin to it). On the other hand, it still forces you to content with `user` being `User | null` anywhere you invoke the `useAuth` hook. That suggests to me we might be using Outlet wrong here, and frankly I'd go to bat against the www on this one too 😅 

I think `Outlet` is fantastic for inserting components across a number of views, like our nav bar, which is how the [React Router documentation suggests its used](https://reactrouter.com/en/main/components/outlet). Because we're forced to content with the types, I think we should focus our solution there instead by recycling a common function naming convention: prefixing with `get` and `find` (`useGetUser` / `useFindUser`). Commonly, `get` isn't expected to fail, it throws an error if it doesn't find what its looking for, but `find` can come back empty handed without it being considered an error. Mapping that idea to our use case:
- Rename `useAuth` to `useFindUser` -> it isn't considered an error if a user isn't found, so it returns `User | null`
- `useGetUser` would use `useFindUser` under the hood to get the user and redirect to `/` if `user` is `null`

Now, in places like the nav bar where we want to render two separate components based on if `user` is defined, we can use `useFindUser`, and in places where we need the component to be restricted to just authenticated users, we can use `useGetUser`. 

Once we've got those types situated, I think we should still still keep `ProtectedRoute`, but maybe shift it up to be wrapping all of the routes that are "protected", instead of wrapping them individually. That will sort out the types + the chance there's a view that doesn't need `user` but isn't meant to be seen by non-users

Reply:
😅 I am not forcing any battles with the www 😂. After office hours on Monday and our talk about how to handle not passing props down a never ending chain, I had a feeling this area would get some tweaks lol. 

This is really helpful to know about the why of the naming convention, I had see both use and kind of was seeing a difference but didn't fully know this! 

So now we have `useAuth` call at the top in `App` and it gets passed along basically, but with this would `Layout` and `App` now have their own calls of the needed function in them? Like `App` using `useGetUser` and passing that to the `ProtectedRoute` and other routes, while `Layout` would no longer get a user prop because it can call `useFindUser` because we need to handle the view of the navbar based on the null vs user?

Reply:
Commenting because I agree with using a ProtectedRoute. We can have it just wrap the routes using Auth.